docs: remove Nginx ingress, add 4.9.a guidance (#10512)
spectrocloud/librarium: docs/docs-content/troubleshooting/palette-upgrade.md

     user-866c7f779d-drf9w                  1/1     Running   0          31m
     user-866c7f779d-rm4hw                  1/1     Running   0          31m
   ```

> ## `configserver` Stuck on `init-rootdomain-traefik` After Upgrade to 4.9.a

Reviewer: I'd put this scenario at the top since it's the newest :)

After upgrading [self-hosted Palette](../enterprise-version/enterprise-version.md) or
[Palette VerteX](../vertex/vertex.md) to version 4.9.a, the `configserver` pod may remain in `Init:CrashLoopBackOff` for
the `init-rootdomain-traefik` init container. This happens when leftover Nginx ingress controller pods on the same nodes
hold host ports 80, 443, and 9090, which prevents the new Traefik DaemonSet pods from scheduling. The automatic Nginx
cleanup script that runs at `configserver` startup cannot run because `configserver` itself never reaches **Ready**, so
the leftover Nginx resources must be removed manually.

> This scenario applies only to Enterprise installs of self-hosted Palette and Palette VerteX.

Reviewer: ```suggestion
```

Redundant, as we specify self-hosted in the opening sentence.

:::warning

Do not delete the `default-ssl-certificate` Secret in the `ingress-nginx` namespace, and do not delete the

> `ingress-nginx` namespace itself. The Enterprise cert-bridge continues to copy your uploaded Transport Layer Security

Reviewer: ```suggestion
`ingress-nginx` namespace itself. The cert-bridge continues to copy your uploaded Transport Layer Security
```

(TLS) certificate from the `default-ssl-certificate` Secret into Traefik on every `configserver` restart. Deleting
either resource breaks ingress TLS.

:::

### Debug Steps

1. Connect to the cluster using the cluster's kubeconfig file. Refer to the
   [Access Cluster with CLI](../clusters/cluster-management/palette-webctl.md) guide for additional guidance.

2. Verify that the `configserver` pod is in the `Init:CrashLoopBackOff` state and inspect the init container logs.

>    ```shell

Reviewer: To keep the output less cluttered, I would put both commands here in the same code block and then both outputs in the same block. At a glance, it's hard to differentiate command from output.

   kubectl get pod --namespace hubble-system --selector component=configserver
   ```

>    ```bash hideClipboard title="Expected output"

Reviewer: ```suggestion
   ```bash hideClipboard title="Example output"
```

I prefer "Example" since the output will slightly vary

   NAME                            READY   STATUS                  RESTARTS   AGE
   configserver-685fcc5b6d-t8f8h   0/1     Init:CrashLoopBackOff   5          8m
   ```

   ```shell
   kubectl logs <configserver-pod> --namespace hubble-system --container init-rootdomain-traefik --previous | tail
   ```

>    ```bash hideClipboard title="Expected output"

Reviewer: ```suggestion
   ```bash hideClipboard title="Example output"
```

   level=fatal msg="not able to get the host ip. got NotFound: Resource 'websecure' not found of type 'pod'"
   ```

3. Confirm that the Traefik DaemonSet pods are stuck in `Pending` because the host ports are unavailable.

   ```shell
   kubectl get pods --namespace ingress-traefik
   ```

>    ```bash hideClipboard title="Expected output"

Reviewer: ```suggestion
   ```bash hideClipboard title="Example output"
```

   NAME                               READY   STATUS    RESTARTS   AGE
   traefik-ingress-controller-dkjs5   0/1     Pending   0          8m
   traefik-ingress-controller-nltlh   0/1     Pending   0          8m
   ```

   If the scheduler events for those pods show `didn't have free ports for the requested pod ports`, leftover Nginx pods
   are holding host ports 80, 443, and 9090 on the same nodes.

4. Delete the cluster-scoped Nginx resources.

   ```shell
   kubectl delete validatingwebhookconfiguration ingress-nginx-admission --ignore-not-found
   kubectl delete clusterrole ingress-nginx ingress-nginx-admission --ignore-not-found
   kubectl delete clusterrolebinding ingress-nginx ingress-nginx-admission --ignore-not-found
   kubectl delete ingressclass nginx --ignore-not-found
   ```

>    ```bash hideClipboard title="Expected output"

Reviewer: ```suggestion
   ```bash hideClipboard title="Example output"
```

   ```

5. Delete the Nginx resources inside the `ingress-nginx` namespace. Do not delete the namespace itself or the
   `default-ssl-certificate` Secret.

   ```shell
   kubectl delete daemonset ingress-nginx-controller --namespace ingress-nginx --ignore-not-found
   kubectl delete service ingress-nginx-controller --namespace ingress-nginx --ignore-not-found
   kubectl delete service ingress-nginx-controller-admission --namespace ingress-nginx --ignore-not-found
   kubectl delete configmap ingress-nginx-controller --namespace ingress-nginx --ignore-not-found
   kubectl delete serviceaccount ingress-nginx ingress-nginx-admission --namespace ingress-nginx --ignore-not-found
   kubectl delete role ingress-nginx ingress-nginx-admission --namespace ingress-nginx --ignore-not-found
   kubectl delete rolebinding ingress-nginx ingress-nginx-admission --namespace ingress-nginx --ignore-not-found
   kubectl delete secret ingress-nginx-admission --namespace ingress-nginx --ignore-not-found
   kubectl delete lease ingress-nginx-leader --namespace ingress-nginx --ignore-not-found
   kubectl delete certificate ingress-nginx-admission --namespace ingress-nginx --ignore-not-found
   ```

>    ```bash hideClipboard title="Expected output"

Reviewer: ```suggestion
   ```bash hideClipboard title="Example output"
```

   daemonset.apps "ingress-nginx-controller" deleted
   service "ingress-nginx-controller" deleted
   service "ingress-nginx-controller-admission" deleted
   configmap "ingress-nginx-controller" deleted
   serviceaccount "ingress-nginx" deleted
   serviceaccount "ingress-nginx-admission" deleted
   role.rbac.authorization.k8s.io "ingress-nginx" deleted
   role.rbac.authorization.k8s.io "ingress-nginx-admission" deleted
   rolebinding.rbac.authorization.k8s.io "ingress-nginx" deleted
   rolebinding.rbac.authorization.k8s.io "ingress-nginx-admission" deleted
   secret "ingress-nginx-admission" deleted
   lease.coordination.k8s.io "ingress-nginx-leader" deleted
   certificate.cert-manager.io "ingress-nginx-admission" deleted
   ```

6. Delete the legacy `nginx-tls` Secret and `nginx` Certificate from each application namespace.

   ```shell
   for ns in hubble-system ui-system cp-system; do
     kubectl delete certificate nginx --namespace "$ns" --ignore-not-found
     kubectl delete secret nginx-tls --namespace "$ns" --ignore-not-found
   done
   ```

>    ```bash hideClipboard title="Expected output"

Reviewer: ```suggestion
   ```bash hideClipboard title="Example output"
```

   certificate.cert-manager.io "nginx" deleted
   secret "nginx-tls" deleted
   certificate.cert-manager.io "nginx" deleted
   secret "nginx-tls" deleted
   certificate.cert-manager.io "nginx" deleted
   secret "nginx-tls" deleted
   ```

> 7. Verify that the cluster-scoped Nginx resources are gone and that the `ingress-nginx` namespace and

Reviewer: Consider breaking this one into two steps (1. verifying the resources are gone and 2. that the namespace is preserved)

   `default-ssl-certificate` Secret are preserved.

   ```shell
   kubectl get validatingwebhookconfiguration ingress-nginx-admission
   kubectl get clusterrole ingress-nginx ingress-nginx-admission
   kubectl get clusterrolebinding ingress-nginx ingress-nginx-admission
   kubectl get ingressclass nginx
   ```

>    ```bash hideClipboard title="Expected output"

Reviewer: ```suggestion
   ```bash hideClipboard title="Example output"
```

   Error from server (NotFound): validatingwebhookconfigurations.admissionregistration.k8s.io "ingress-nginx-admission" not found
   Error from server (NotFound): clusterroles.rbac.authorization.k8s.io "ingress-nginx" not found
   Error from server (NotFound): clusterrolebindings.rbac.authorization.k8s.io "ingress-nginx" not found
   Error from server (NotFound): ingressclasses.networking.k8s.io "nginx" not found
   ```

   ```shell
   kubectl get namespace ingress-nginx
   kubectl get secret default-ssl-certificate --namespace ingress-nginx
   ```

>    ```bash hideClipboard title="Expected output"

Reviewer: ```suggestion
   ```bash hideClipboard title="Example output"
```

   NAME            STATUS   AGE
   ingress-nginx   Active   42d

   NAME                      TYPE                DATA   AGE
   default-ssl-certificate   kubernetes.io/tls   2      42d
   ```

8. Once the leftover Nginx pods are removed, the scheduler places the Traefik DaemonSet pods, `configserver` reaches

>    **Ready**, and the remaining automatic cleanup completes on the next `configserver` start. Confirm that the Traefik

Reviewer: ```suggestion
   **Ready**, and the remaining automatic cleanup processes complete on the next `configserver` start. Confirm that the Traefik
```

   pods are running and that `configserver` is **Ready**.

   ```shell
   kubectl get pods --namespace ingress-traefik
   kubectl get pod --namespace hubble-system --selector component=configserver
   ```

>    ```bash hideClipboard title="Expected output"

Reviewer: ```suggestion
   ```bash hideClipboard title="Example output"
```
